src/hammer-vlsi/cli_driver_test.py

```diff
@@ -39,6 +42,199 @@ def generate_dummy_config(self, syn_rundir: str, config_path: str, top_module: s
 
         return config
 
+    def run_syn_to_par_with_output(self, config_path: str, syn_rundir: str,
+                                   par_rundir: str,
+                                   syn_out_path: str,
+                                   syn_to_par_out_path: str) -> None:
+        # Check that running the CLIDriver executes successfully (code 0).
+        with self.assertRaises(SystemExit) as cm:  # type: ignore
+            CLIDriver().main(args=[
+                "syn",  # action
+                "-p", config_path,
+                "--output", syn_out_path,
+                "--syn_rundir", syn_rundir,
+                "--par_rundir", par_rundir
+            ])
+        self.assertEqual(cm.exception.code, 0)
+        # Now run syn-to-par with the main config as well as the outputs.
+        with self.assertRaises(SystemExit) as cm:  # type: ignore
+            CLIDriver().main(args=[
+                "syn-to-par",  # action
+                "-p", config_path,
+                "-p", syn_out_path,
+                "--output", syn_to_par_out_path,
+                "--syn_rundir", syn_rundir,
+                "--par_rundir", par_rundir
+            ])
+        self.assertEqual(cm.exception.code, 0)
+
+    def test_syn_to_par(self) -> None:
+        """
+        Test that syn-to-par works with the output-only JSON.
+        """
+
+        # Set up some temporary folders for the unit test.
+        syn_rundir = tempfile.mkdtemp()
+        par_rundir = tempfile.mkdtemp()
+
+        # Generate a config for testing.
+        top_module = "dummy"
+        config_path = os.path.join(syn_rundir, "run_config.json")
+        syn_out_path = os.path.join(syn_rundir, "syn_out.json")
+        syn_to_par_out_path = os.path.join(syn_rundir, "syn_par_out.json")
+        self.generate_dummy_config(syn_rundir, config_path, top_module)
+
+        self.run_syn_to_par_with_output(config_path, syn_rundir, par_rundir,
+                                        syn_out_path, syn_to_par_out_path)
+
+        # synthesis output should NOT keep other settings
+        with open(syn_out_path, "r") as f:
+            syn_output = json.loads(f.read())
+            self.assertEqual(syn_output["synthesis.outputs.output_files"], [])
```

Comment:
Isn't this one of the few things that should be in the syn_output?

Reply:
It is in syn_output. The comment for "other settings" refers to settings like `vlsi.core.technology`.